components/draggable/index.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * WordPress Dependencies
+ */
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal Dependencies
+ */
+import withSafeTimeout from '../higher-order/with-safe-timeout';
+import './style.scss';
+
+const dragImageClass = 'components-draggable__invisible-drag-image';
+const cloneWrapperClass = 'components-draggable__clone';
+const cloneHeightTransformationBreakpoint = 700;
+const clonePadding = 20;
+
+class Draggable extends Component {
+	constructor() {
+		super( ...arguments );
+		this.onDragStart = this.onDragStart.bind( this );
+		this.onDragOver = this.onDragOver.bind( this );
+		this.onDragEnd = this.onDragEnd.bind( this );
+	}
+
+	/**
+	 * Removes the element clone, resets cursor, and removes drag listener.
+	 * @param  {Object} event     The non-custom DragEvent.
+	 */
+	onDragEnd( event ) {
+		const { onDragEnd = noop } = this.props;
+		this.removeDragClone();
+		// Reset cursor.
+		document.body.classList.remove( 'dragging' );
+		event.preventDefault();
+
+		this.props.setTimeout( onDragEnd );
+	}
+
+	/*
+	 * Updates positioning of element clone based on mouse movement during dragging.
+	 * @param  {Object} event     The non-custom DragEvent.
+	 */
+	onDragOver( event ) {
+		this.cloneWrapper.style.top =
+			`${ parseInt( this.cloneWrapper.style.top, 10 ) + event.clientY - this.cursorTop }px`;
+		this.cloneWrapper.style.left =
+			`${ parseInt( this.cloneWrapper.style.left, 10 ) + event.clientX - this.cursorLeft }px`;
+
+		// Update cursor coordinates.
+		this.cursorLeft = event.clientX;
+		this.cursorTop = event.clientY;
+	}
+
+	/**
+	 *  - Clones the current element and spawns clone over original element.
+	 *  - Adds a fake temporary drag image to avoid browser defaults.
+	 *  - Sets transfer data.
+	 *  - Adds dragover listener.
+	 * @param  {Object} event					The non-custom DragEvent.
+	 * @param  {string} elementId				The HTML id of the element to be dragged.
+	 * @param  {Object} transferData			The data to be set to the event's dataTransfer - to be accessible in any later drop logic.
+	 */
+	onDragStart( event ) {
+		const { elementId, transferData, onDragStart = noop } = this.props;
+		const element = document.getElementById( elementId );
+		const elementWrapper = element.parentNode;
+		const elementRect = element.getBoundingClientRect();
+		const elementTopOffset = parseInt( elementRect.top, 10 );
+		const elementLeftOffset = parseInt( elementRect.left, 10 );
+		const clone = element.cloneNode( true );
+		const dragImage = document.createElement( 'div' );
+
+		// Set a fake drag image to avoid browser defaults. Remove from DOM right after.
+		if ( 'function' === typeof event.dataTransfer.setDragImage ) {
```

Comment:
Why do we check `typeof` ?

_...And as a follow-up, should we note via code comment that it's for browser compatibility?_

Reply:
Of course - will do.